cmd/sonobuoy/app/gen_test.go

```diff
@@ -244,44 +257,105 @@ func TestResolveConfig(t *testing.T) {
+				},
+			},
+		}, {
+			name:  "Plugins loaded by directory and by name",
+			input: "--plugin e2e --plugin testdata/testPluginDir --plugin testdata/testPluginDir/pluginNotYAML.ext",
+			expected: &config.Config{
+				Namespace:        "sonobuoy",
+				WorkerImage:      "gcr.io/heptio-images/sonobuoy:" + buildinfo.Version,
+				ImagePullPolicy:  "IfNotPresent",
+				PluginSelections: nil,
+				PluginSearchPath: defaultPluginSearchPath,
+				Aggregation:      defaultAggr,
+				Resources:        config.DefaultResources,
+			},
+			expectedGenConfigPlugins: &client.GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{SonobuoyConfig: manifest.SonobuoyConfig{Driver: "Job", PluginName: "plugin1", SkipCleanup: false, ResultFormat: "raw"}, Spec: manifest.Container{Container: v1.Container{Name: "plugin", Image: "foo/bar:v1", Command: []string{"./run.sh"}, VolumeMounts: []v1.VolumeMount{{ReadOnly: false, Name: "results", MountPath: plugin.ResultsDir}}}}},
+					{SonobuoyConfig: manifest.SonobuoyConfig{Driver: "Job", PluginName: "plugin2", SkipCleanup: false, ResultFormat: "raw"}, Spec: manifest.Container{Container: v1.Container{Name: "plugin", Image: "foo/bar:v2", Command: []string{"./run.sh"}, VolumeMounts: []v1.VolumeMount{{ReadOnly: false, Name: "results", MountPath: plugin.ResultsDir}}}}},
+					{SonobuoyConfig: manifest.SonobuoyConfig{Driver: "Job", PluginName: "plugin3", SkipCleanup: false, ResultFormat: "raw"}, Spec: manifest.Container{Container: v1.Container{Name: "plugin", Image: "foo/bar:v3", Command: []string{"./run.sh"}, VolumeMounts: []v1.VolumeMount{{ReadOnly: false, Name: "results", MountPath: plugin.ResultsDir}}}}},
+				},
+				DynamicPlugins: []string{"e2e"},
+			},
 		},
 	}
 
 	for _, tc := range tcs {
 		t.Run(tc.name, func(t *testing.T) {
-			// Simulate parsing of input via CLI. Making this optional to avoid complicating
-			// setup for other tests which just explicitly set the values.
-			if len(tc.cliInput) > 0 {
-				fs := GenFlagSet(tc.input, EnabledRBACMode)
-				if err := fs.Parse(strings.Split(tc.cliInput, " ")); err != nil {
-					t.Fatalf("Failed to parse CLI input %q: %v", tc.cliInput, err)
+			// Write and load a config file
+			if len(tc.configFileContents) > 0 {
+				tmpFile, err := ioutil.TempFile("", "")
+				if err != nil {
+					t.Fatalf("Failed to create temp file for test: %v", err)
+				}
+				defer os.Remove(tmpFile.Name())
+				err = ioutil.WriteFile(tmpFile.Name(), []byte(tc.configFileContents), 0666)
+				if err != nil {
+					t.Fatalf("Failed to write temp config file for test: %v", err)
 				}
+				tc.input = strings.Replace(tc.input, dynamicConfigFileName, tmpFile.Name(), -1)
 			}
 
-			conf := tc.input.resolveConfig()
```

Comment:
The old test was using `resolveConfig` which generates the `config.Config` object. However, this isn't sufficient to see if the GenConfig would be appropriate so we were missing the ability to test large sections of input processing.